torchtune/training/_activation_offloading.py

```diff
@@ -281,8 +285,21 @@ def wait_and_del_remaining_references() -> None:
                 def hook(outputs, inputs):
                     # create events for the current node inputs/outputs if they were streamed in
                     if brought_back_from_cpu:
-                        event = self.s0.record_event()
-                        self.bwd_ev_stash[unpack_tensor_id] = event
+                        # if any of the outputs is a view of the tensor, meaning the tensor might be used later,
+                        # we cannot presume to delete it after only the current node is done! So we use our frenemy,
+                        # record_stream, to ensure the Tensor stays unmessed with until it's done getting used
+                        # in the compute stream (s0 here). Note that the con here is we introduce non-deterministic
+                        # memory usage, but this case should not happen often.
+                        unpacked_tensor = self.bwd_tensor_stash[unpack_tensor_id]
+                        if any(
+                            o.untyped_storage() is unpacked_tensor.untyped_storage()
```

Comment:
When I run with this, I get:

```
    o.untyped_storage() is unpacked_tensor.untyped_storage()
AttributeError: 'NoneType' object has no attribute 'untyped_storage'
```

I think `o` _can_ be None.

Reply:
Good catch thanks! Should be updated in latest commit